tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -662,4 +662,25 @@ function( $data, $attachment_id ) {
 		$this->assertNotContains( 'image/invalid', $mime_types );
 		$this->assertSame( array( 'image/jpeg', 'image/webp' ), $mime_types );
 	}
+
+	/**
+	 * Prevent replacing an image if image was uploaded via external source or plugin.
+	 *
+	 * @group webp_uploads_update_image_references
+	 *
+	 * @test
+	 */
+	public function it_should_prevent_replacing_an_image_uploaded_via_external_source() {
+		add_filter(
```

Comment:
We need to call `remove_all_filters( 'webp_uploads_pre_replace_additional_image_source' )` before adding a new hook to make sure that no other hooks are registered for this filter.

Reply:
Thanks @eugene-manuilov I have added changes according to your feedback now.